### ISSUE TITLE
Support older Java versions

### ISSUE THREAD

**Issue by 190n** [alda-lang/alda#171]
Alda isn't working for me with OpenJDK.

OS: Debian 8

Java (`$ java -version`):

```
java version "1.7.0_91"
OpenJDK Runtime Environment (IcedTea 2.6.3) (7u91-2.6.3-1~deb8u1)
OpenJDK 64-Bit Server VM (build 24.91-b01, mixed mode)
```

I get this error:

```
Exception in thread "main" java.lang.UnsupportedClassVersionError: alda/Client : Unsupported major.minor version 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:482)
```

I have Oracle's JVM installed outside my PATH for playing Minecraft, so I can just set up an alias (`alias alda='/usr/local/java/bin/java -jar ~/.local/bin/alda $@`). However, it'd be nice to use the open-source version. Anyone know what's going on?

**Comment by crisptrutski** [comment]
This sounds like a 1.7 aka 51.0 (your OpenJDK version) vs 1.8  aka 52.0 (what Alda seems to be compiled for)

Perhaps Alda can be cross compiled for 1.7 compatibility?

**Comment by daveyarwood** [comment]
@crisptrutski Yep, that's probably the issue. We should just be able to add cross-compilation options to the Boot `javac` task, no?

I might be wrong about this, but I think Alda ought to be compatible with JDKs as far back as 1.6. When I have a minute, I'll look into cross-compiling to support JDK 1.6+; I think that should support most users at this point.

**Comment by 190n** [comment]
Thanks for the quick reply! Should I try to compile it myself, or not
bother?

**Comment by daveyarwood** [comment]
@190n you're welcome to play around with it -- I won't have a chance to try myself until tonight/tomorrow probably.

To build:

1. Install [Boot](http://boot-clj.com).
2. Clone this repo.
3. In the project directory, run `boot build -o /tmp`. This will build the Java/Clojure source code according to the `build.boot` file, and create executables in `/tmp` (or substitute the directory of your choice)

The thing to add would be `javac` options, which would go in here: https://github.com/alda-lang/alda/blob/master/build.boot#L133-L136

e.g. `(javac :options "-source 1.6")`

We'll want to do something like this: http://docs.oracle.com/javase/7/docs/technotes/tools/windows/javac.html#crosscomp-example

**Comment by daveyarwood** [comment]
Fixed in 1.0.0-rc3, which you can download here: https://github.com/alda-lang/alda/releases/tag/1.0.0-rc3

Alda should now work as long as you have Java version 1.7+

**Comment by 190n** [comment]
Thanks!
